cpp/CMakeLists.txt

```diff
@@ -861,6 +857,12 @@ if(ARROW_USE_XSIMD)
   list(APPEND ARROW_STATIC_LINK_LIBS ${ARROW_XSIMD})
 endif()
 
+if(ARROW_WITH_PROTOBUF)
```

Comment:
```suggestion
# This should be done after if(ARROW_ORC) and if(ARROW_WITH_OPENTELEMETRY)
# because they depend on Protobuf.
if(ARROW_WITH_PROTOBUF)
```